src/os/FileStore.cc

```diff
@@ -1830,13 +1840,65 @@ void FileStore::queue_op(OpSequencer *osr, Op *o)
   dout(5) << "queue_op " << o << " seq " << o->op
 	  << " " << *osr
 	  << " " << o->bytes << " bytes"
-	  << "   (queue has " << throttle_ops.get_current() << " ops and " << throttle_bytes.get_current() << " bytes)"
+	  << "   (queue has " << throttle_ops.get_current() << " ops and " 
+          << throttle_bytes.get_current() << " bytes)"
 	  << dendl;
   op_wq.queue(osr);
+
+  if (g_conf->journal_dynamic_throttle) {
+  /* Trying to throttle the journal commit based on the number of
+     outstanding bytes to be applied. This will restrict the amount of
+     memory growth in the write path. The mode of throttle is to introduce
+     an incremental delay based on the % increase above an outstanding byte
+     threshold.
+   */
+    uint64_t outstanding_bytes = m_filestore_current_queue_bytes.read(); 
+    bool induce_delay = false;
+    uint32_t percentage_more = 0;
+    
+    if (outstanding_bytes > m_filestore_queue_max_bytes) {
+      percentage_more = (uint32_t)((100* (outstanding_bytes -
```

Comment:
Do this in terms of a double, casting this to a uint32_t doesn't relaly make sense, calc_wait_time can just take a double instead.